streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/PublisherExtensions.kt

```diff
@@ -97,3 +109,11 @@ fun <T> Publisher<T>.startWith(value: T): Publisher<T> {
 fun <T, R> Publisher<T>.filterNotNull(block: ((T) -> R?)): Publisher<R> {
     return this.filter { block(it) != null }.map { block(it)!! }
 }
+
+@ExperimentalTime
+fun <T> Publisher<T>.debounce(
+    delayMs: Duration,
```

Comment:
It would be great is you have an interface where you can specify a time unit instead of always requiring a conversion to milliseconds.

Is there something akin to `TimeUnit` in Kotlin?

Reply:
`Duration` does exactly this. You can pass `1.minute` or `10.seconds`. I think we should rename `delayMs` to `delay`

Reply:
FYI https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.time/-duration/index.html

Reply:
Also, should it be `timeout` instead of `delay`?

Reply:
I don't care, as long as its not delayMS (as Duration is unit unaware). Documentation says Timespan

Reply:
Renamed `timeout` like http://reactivex.io/RxJava/javadoc/rx/Observable.html#debounce-long-java.util.concurrent.TimeUnit-